### DISCUSSION TITLE
Pruning after emptying manifest repo

### DISCUSSION THREAD

**Discussion by mikesir87** [fluxcd/flux2#2156] [Q&A] [answered]
When a `Kustomization` points to a repo and no manifests are found, it gives an error (understandably). Unfortunately, it seems like when that happens, no pruning occurs. So, if a repo that once had manifests is emptied out, the last manifests/objects are not pruned.

We run into this fairly often for demo repos where we test things out before a presentation and then clear things out before the actual presentation. Since the last commit removes objects without pruning, we have to manually go and remove the objects.

Is there another process/flag I’m potentially missing?

**Comment by stefanprodan** [comment, marked as answer]
> When a Kustomization points to a repo and no manifests are found, it gives an error (understandably).

This was the case many months ago, please upgrade to latest Flux.

**Comment by mikesir87** [comment]
Doh! I had misread the version number we were running, so had thought we were at the latest. I'll go ahead and upgrade. Thanks @stefanprodan!
